app/uk/gov/hmrc/personaldetailsvalidation/services/RepoControlService.scala

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.personaldetailsvalidation.services
+
+import akka.Done
+import cats.data.EitherT
+import play.api.Logging
+import uk.gov.hmrc.http.{HeaderCarrier, SessionId}
+
+import java.time.LocalDateTime
+import javax.inject.{Inject,Singleton}
+import uk.gov.hmrc.personaldetailsvalidation.model._
+import uk.gov.hmrc.crypto.PlainText
+
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class RepoControlService @Inject()(pdvService: PersonalDetailsValidatorService, associationService: AssociationService) extends Logging{
+
+  def insertPDVAndAssociationRecord(personalDetailsValidation: PersonalDetailsValidation,
+                                    opCredID: Option[String],
+                                    headerCarrier: HeaderCarrier)
+                                   (implicit ec: ExecutionContext, encryption: Encryption): EitherT[Future, Exception, Done] = {
```

Comment:
if we pass in the encryption rather than inject it in RepoControlService. could this lead to people abusing this method call and passing in a different instance of Encryption

Reply:
i don't know for certain but it is the same format as FPC

Reply:
👍